include/openthread/mdns.h

```diff
@@ -388,6 +406,88 @@ otError otMdnsRegisterKey(otInstance            *aInstance,
  */
 otError otMdnsUnregisterKey(otInstance *aInstance, const otMdnsKey *aKey);
 
+/**
+ * Allocates a new iterator.
+ *
+ * An allocated iterator must be freed by the caller using `otMdnsFreeIterator()`.
+ *
+ * @param[in] aInstance    The OpenThread instance.
+ *
+ * @returns A pointer to the allocated iterator, or `NULL` if fails to allocate.
+ *
+ */
+otMdnsIterator *otMdnsAllocateIterator(otInstance *aInstance);
+
+/**
+ * Frees a previously allocated iterator.
+ *
+ * @param[in] aInstance    The OpenThread instance.
+ * @param[in] aIterator    The iterator to free.
+ *
+ */
+void otMdnsFreeIterator(otInstance *aInstance, otMdnsIterator *aIterator);
+
+/**
+ * Iterates over registered host entries.
+ *
+ * On success, @p aHost is populated with the next host information. Pointers within the `otMdnsHost` structure
+ * (like `mName`) remain valid until the next call to any OpenThread stack public or platform API/callback.
```

Comment:
Please clarify a little bit what is being described in line 434.

Reply:
Yes. I changed it "OpenThread stack's public or platform API/callback".



---------


Some background on this (for your info):
OpenThread employs a layered architecture to provide flexibility and clear separation of responsibilities:
- Public APIs (`include/openthread/*.h`): These APIs allow upper layers of the software stack to interact with and control OpenThread's behavior.
- Platform APIs (`include/openthread/platform/*.h`): OpenThread relies on these APIs, which must be implemented by the platform layer, to interface with hardware and system-specific features.